docs/concepts/registries.md

```diff
@@ -17,7 +17,7 @@ public void registerBlocks(RegistryEvent.Register<Block> event) {
 }
 ```
 
-The order in which `RegistryEvent.Register` events fire is arbitrary, with the exception that `Block` will *always* fire first, and `Item` will *always* fire second, right after `Block`. After the `Register<Block>` event has fired, all [`ObjectHolder`][ObjectHolder] annotations are refreshed, and after `Register<Item>` has fired they are refreshed again. They are refreshed for a third time after *all* of the other `Register` events have fired.
+The order in which `RegistryEvent.Register` events fire is lexicographically, with the exception that `Block` will *always* fire first, and `Item` will *always* fire second, right after `Block`. After the `Register<Block>` event has fired, all [`ObjectHolder`][ObjectHolder] annotations are refreshed, and after `Register<Item>` has fired they are refreshed again. They are refreshed for a third time after *all* of the other `Register` events have fired.
```

Comment:
It isn't technically alphabetical, though :P I do agree that (since most registry names should consist of just letters) "lexicographical" would confuse more people than it would help, however.